consensus/src/experimental/README.md

```diff
@@ -0,0 +1,122 @@
+---
+id: consensus-decoupled-execution
+title: Decoupled-Execution
+custom_edit_url: https://github.com/diem/diem/edit/main/consensus/src/experimental/README.md
+---
+
+The decoupled execution mod implements the decoupled execution feature of DiemBFT protocol.
+
+> This is an experimental feature, please use it with caution.    
+
+# Overview
+
+In this mod, we implemented decoupled execution. DiemBFT decouples transaction executing and committing from the consensus critical path. It builds an asynchronous pipeline where an ordered block gets processed by a dedicated
+executing thread and a dedicated committing thread. This yields a better throughput as the pipelining
+splits the bottleneck into smaller parts. To tolerate execution uncertainties like bit flips, DiemBFT let the
+nodes agree on the execution result. The nodes asynchronously vote on the execution result. Note that, this
+extra roundtrip is not piggybacked on the consensus communication (HotStuff chains), therefore, it does not
+adversely affect the high throughput and low latency. To help slower nodes catch up, DiemBFT nodes also
+share aggregated signatures on executed results.
+
+# Diagram
+
+Recall that in the DiemBFT protocol, every block goes through a number of notorizations before become decided.
+Fortunately, the blocks form a pipeline of chain rule. 
+On average, each round (that is not a timeout) commits a new block.
+Therefore, the throughput can be roughly estimated as 1/(the time per round). 
+![Chain rule in DiemBFT](docs/chain.png)
+If we look closer to the implementation (shown above), each step consists of executing the block, voting the block, 
+and (after a while) committing (persisting) the block. All of these steps are on the critical path.
+![Chain rule in DiemBFT](docs/before.png)
+Therefore, the throughput is at most 1/(the time of preparation + the time of execution + the time of voting + the time of committing/persisting).
+
+Decoupled execution moves execution and persisting to dedicated threads/tasks. The time cost of execution and committing/persisting
+are no longer on the critical path of the main loop.
+![Chain rule in DiemBFT](docs/after.png)
```

Comment:
for this graph I think we should be more explicit about the proposal -> vote -> ordered and make the other two tasks vertically with the ordering task so that readers can get it intuitively.